conans/model/conf.py

```diff
@@ -47,117 +47,391 @@ def _is_profile_module(module_name):
+
+    def __bool__(self):
+        return bool(self._values)
+
+    __nonzero__ = __bool__
+
+    def __repr__(self):
+        return "Conf: " + repr(self._values)
+
+    def __eq__(self, other):
+        """
+        :type other: Conf
+        """
+        return other._values == self._values
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
 
     def __getitem__(self, name):
-        return self._values.get(name)
+        """
+        DEPRECATED: it's going to disappear in Conan 2.0. Use self.get() instead.
+        """
+        # FIXME: Keeping backward compatibility
+        return self.get(name)
 
     def __setitem__(self, name, value):
-        if name != name.lower():
-            raise ConanException("Conf '{}' must be lowercase".format(name))
-        self._values[name] = value
+        """
+        DEPRECATED: it's going to disappear in Conan 2.0.
+        """
+        # FIXME: Keeping backward compatibility
+        self.define(name, value)  # it's like a new definition
 
     def __delitem__(self, name):
-        del self._values[name]
-
-    def __repr__(self):
-        return "Conf: " + repr(self._values)
+        """
+        DEPRECATED: it's going to disappear in Conan 2.0.
+        """
+        # FIXME: Keeping backward compatibility
+        self.pop(name)
 
     def items(self):
-        return self._values.items()
-
-    def filter_user_modules(self):
-        result = Conf()
+        # FIXME: Keeping backward compatibility
         for k, v in self._values.items():
-            if _is_profile_module(k):
-                result._values[k] = v
-        return result
+            yield k, v.value
 
-    def update(self, other):
+    @property
+    def sha(self):
+        # FIXME: Keeping backward compatibility
+        return self.dumps()
+
+    def get(self, conf_name, default=None):
         """
-        :param other: has more priority than current one
-        :type other: Conf
+        Get all the values belonging to the passed conf name. By default, those values
+        will be returned as a str-like object.
+        """
+        conf_value = self._values.get(conf_name)
+        if conf_value:
+            return conf_value.value
+        else:
+            return default
+
+    def pop(self, conf_name, default=None):
+        """
+        Remove any key-value given the conf name
         """
-        self._values.update(other._values)
+        value = self.get(conf_name, default)
+        self._values.pop(conf_name, None)
+        return value
+
+    def _get_conf_value(self, name, value):
+        """
+        Get a valid _ConfValue object based on the built-in Conf type declared
+        or the value-like object. For now, we only manage lists or strings.
+        """
+        if (name in self._values and isinstance(self._values[name], _ConfListValue)) \
+           or BUILT_IN_CONFS.get(name, [None])[0] is list or isinstance(value, list):
```

Comment:
This is quite complex to understand, for me it is a symptom of the above decision to define and try to validate types.